portfolio/src/main/webapp/discussionScript.js

```diff
@@ -13,16 +19,87 @@ function displayList() {
       content.classList.add("post-content");
       content.textContent = data[i].content;
 
+      let replies = prepareReplies(data[i].replies);
+      replies.classList.add("post-replies");
 
       let singlePost = document.createElement('div');
       singlePost.classList.add("post");
 
+      let replyButton = document.createElement('button');
+      replyButton.textContent = "Reply to this post";
+
+      singlePost.appendChild(userIdentification);
       singlePost.appendChild(title);
       singlePost.appendChild(content);
+      singlePost.appendChild(replies);
+      singlePost.appendChild(replyButton);
 
       posts.appendChild(singlePost);
+
+      replyButton.addEventListener("click", () => {
+        replyButton.classList.add("hidden");
+        createReplyForm(singlePost, data[i].id);
+      });
     }
 
     document.getElementById('discussion-container').appendChild(posts);
   })
 }
+
+/**
+ *  This function creates a form that lets user reply to the comment of their choosing
+ *  and then displays it underneath the post.
+ *  @param {Object} singlePost Post object that is being replied to
```

Comment:
If im reading this right, this should be of type {!Element}

The ! indicates that it can't be null.  This isn't used for primitive types (number, string), which are assumed not to be null, but for any other type, you should use ! to indicate it must be there, or ? to indicate that it could be null

Reply:
that's great. I wonder why they don't have that on their official site.